bson/_cbsonmodule.c

```diff
@@ -1743,6 +1743,42 @@ int write_dict(PyObject* self, buffer_t buffer,
         while (PyDict_Next(dict, &pos, &key, &value)) {
             if (!decode_and_write_pair(self, buffer, key, value,
                                     check_keys, options, top_level)) {
+                if (PyErr_Occurred()) {
+                    PyObject *etype, *evalue, *etrace;
+                    PyErr_Fetch(&etype, &evalue, &etrace);
+                    PyObject *InvalidDocument = _error("InvalidDocument");
```

Comment:
There's a lot of missing error/NULL checks in this code. I'm not comfortable merging this for 4.9 so it will need to wait for the next release.

Reply:
@blink1073 @ShaneHarvey  I took context from code written in the same file, would appreciate if can you point out some existing code from where i can see what all error/Nulls check to put here?

Reply:
Hi @navjots18, apologies for the delay.  I think all we're missing in this block is initializing the `etype`, `evalue`, and `etrace` to NULL as we do [here](https://github.com/mongodb/mongo-python-driver/blob/2f1227c504064d6859cb05d567999262f99b7937/bson/_cbsonmodule.c#L2515).  Otherwise I agree this matches the rest of the existing code block.

Reply:
Hey @blink1073 fixed these issues, can you check again?